oggm/utils/_downloads.py

```diff
@@ -199,6 +230,42 @@ def _cached_download_helper(cache_obj_name, dl_func, reset=False):
     return cache_path
 
 
+def _verified_download_helper(cache_obj_name, dl_func, reset=False):
+    """Helper function for downloads.
+
+    Verifies the size and hash of the downloaded file against the included
+    list of known static files.
+    Uses _cached_download_helper to perform the actual download.
+    """
+    path = _cached_download_helper(cache_obj_name, dl_func, reset)
+
+    if cfg.PARAMS['dl_verify']:
+        data = get_dl_verify_data()
+        sha256 = hashlib.sha256()
+        with open(path, 'rb') as f:
+            while True:
+                b = f.read(0xFFFF)
+                if not b:
+                    break
+                sha256.update(b)
+        size = os.path.getsize(path)
+
+        if cache_obj_name not in data:
+            logger.warning('No known hash for %s: %s %s' %
+                           (path, size, sha256.hexdigest()))
+        else:
+            data = data[cache_obj_name]
+            if data[0] != size or data[1] != sha256.digest():
+                err = '%s failed to verify!\nis: %s %s\nexpected: %s %s' %
```

Comment:
`err = '%s failed to verify!\nis: %s %s\nexpected: %s %s' % (`
 ` ` ` ` ` ` ` ` `path, size, sha256.hexdigest(), data[0], data[1].hex())`

should work